client/components/data/query-active-promotions/index.jsx

```diff
@@ -1,45 +1,25 @@
-/** @format */
-
 /**
  * External dependencies
  */
-
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { isRequestingActivePromotions } from 'state/active-promotions/selectors';
 import { requestActivePromotions } from 'state/active-promotions/actions';
 
-class QueryActivePromotions extends Component {
-	UNSAFE_componentWillMount() {
-		if ( ! this.props.requestingActivePromotions ) {
-			this.props.requestActivePromotions();
+export default function QueryActivePromotions() {
+	const requestingActivePromotions = useRef( useSelector( isRequestingActivePromotions ) );
+	const dispatch = useDispatch();
+
+	// Only runs on mount.
+	useEffect( () => {
+		if ( ! requestingActivePromotions.current ) {
+			dispatch( requestActivePromotions() );
 		}
-	}
+	}, [ dispatch ] );
```

Comment:
Do you agree with including `dispatch` in the dependency array, even though it should never change, to avoid linting errors with the hook rules? Would it be better to use an empty array and an `eslint` disabling comment?

Reply:
It can change in unlikely cases, e.g., when you rerender the `<Provider>` with a different `store`. It's probably right to include it, although it's a bit annoying to look at the code 🙃

Reply:
Thank you, I'll keep it!